### PR TITLE
Add logr.LogSink wrapper and new methods Error, Debug

### PR DESCRIPTION
**Pull request by tfussell**
This PR adds a new type, `LogrSink` which adapts the `micrologger.Logger` interface to the `logr.LogSink` interface so it can be passed into `klog` and used as a sink with the `controller-runtime` deferred logger. It also adds a few methods to the `Logger` interface for completeness.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -84,13 +109,21 @@ func (l *MicroLogger) LogCtx(ctx context.Context, keyVals ...interface{}) {
 	l.log(keyValsWithMeta(ctx, keyVals))
 }
 
-func (l *MicroLogger) With(keyVals ...interface{}) Logger {
-	keyVals = processStack(keyVals)
+func (l *MicroLogger) deepCopy() *MicroLogger {
 	return &MicroLogger{
-		logger: kitlog.With(l.logger, keyVals...),
+		info:      l.info,
+		logger:    l.logger,
```
Comment (tfussell) [inline logger.go:115]: This isn't technically a deep copy, but `go-kit/log` appears to be immutable so I think it's close enough.

File: activation_logger.go [hunk 2]
```diff
@@ -84,18 +84,34 @@ func NewActivation(config ActivationLoggerConfig) (Logger, error) {
...
+func (l *activationLogger) Errorf(ctx context.Context, err error, format string, params ...interface{}) {
+	l.Error(ctx, err, fmt.Sprintf(format, params...))
+}
+
+func (l *activationLogger) Info(ctx context.Context, message string) {
+	l.LogCtx(ctx, "level", "debug", "message", message)
+}
+
+func (l *activationLogger) Infof(ctx context.Context, format string, params ...interface{}) {
+	l.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf(format, params...))
```
Comment (TheoBrigitte) [inline activation_logger.go:112]: Feels weird to have `level=debug` when using `Info*` methods.

**Review by TheoBrigitte — COMMENTED** [review]
Looks overall reasonable to me

File: activation_logger.go (hunk 2 above)
Comment (tfussell) [reply activation_logger.go:112]: Oh, that's a copy-and-paste mistake :)

**Comment by tfussell** [comment]
> I'd argue `Info*` is not that useful. We consciously made the decision to not have it in the past: [#60 (comment)](https://github.com/giantswarm/micrologger/pull/60#discussion_r532725012)
> 
> I still think that way.
> 
> I'm fine with `Debug` and `Error` but I think they only clutter the interface. There is no real gain of using them over their `*f` siblings.

I can think of a few small advantages of these functions:
- Avoid overhead when printing plain strings. `golangci-lint` for example warns if you use `fmt.Sprintf` with a plain string as there's no reason not to just use the string directly.
- Avoid some possible attack vectors if printing untrusted input https://owasp.org/www-community/attacks/Format_string_attack.
- IntelliJ doesn't know about the format string in our custom `*f` functions so I like to use `fmt.Sprintf` to get syntax highlighting and type checking.
- Most libraries that expose `*f` functions also expose plain functions so this makes the library consitent with developer expectations.

File: activation_logger.go (hunk 2 above)
Comment (tfussell) [reply activation_logger.go]: Fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add `Info` and `Infof` methods which log a message or a format string
+  with parameters (respectively) at the "info" level.
```
Comment (tfussell) [inline CHANGELOG.md:15]:
```suggestion
```

**Comment by tfussell** [comment]
> I'd argue `Info*` is not that useful. We consciously made the decision to not have it in the past: [#60 (comment)](https://github.com/giantswarm/micrologger/pull/60#discussion_r532725012)

`logr.LogSink` requires two logging methods to be implemented, `Error` and `Info`. I initially felt that an additional "info" level would be helpful to support this interface, but I guess it works mapping "info" to "debug". I've removed them.

**Review by MarcelMue — APPROVED** [review]
Makes sense to me

**Review by kopiczko — APPROVED** [review]

**Review by yulianedyalkova — APPROVED** [review]

**Pull request merged**